### PR TITLE
Add nonce cookie validation to login flow.

### PR DESCRIPTION
**Pull request by inthirakumaaran**
Introduce nonce cookie validation for login flow.

Add the following config to the deployment .toml in order to use the feature.
```
[session.nonce.cookie]
enabled="true"
```

### Depended on PR
https://github.com/wso2/product-is/pull/9926

### CODE REVIEW

**Comment by inthirakumaaran** [comment]
Let's merge the PR https://github.com/wso2/product-is/pull/9926 first and then try the PR builder again before merging this.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     * Get dynamic name for the nonce cookie
+     *
+     * @param context Authentication Context.
+     *
+     * @return name of the nonce cookie for that context.
+     */
+    public static String getNonceCookieName(AuthenticationContext context) {
+
+        String cookieName = NONCE_COOKIE + "-" + context.getContextIdentifier();
+        return cookieName;
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java:51]: no need to declare a variable.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * Check whether nonce cookie config is enabled or not.
+     *
+     * @return nonce cookie enabled or not.
+     */
+    public static boolean isNonceCookieEnabled() {
+
+        if (nonceCookieConfig == null) {
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java:127]:
```suggestion
        if (nonceCookieConfig == null) {
            nonceCookieConfig = Boolean.parseBoolean(IdentityUtil.getProperty(NONCE_COOKIE_CONFIG));
        }
        return nonceCookieConfig;
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                nonceFromSession = nonceCookie.getValue();
+            }
+            if (!StringUtils.isEmpty(nonceFromContext) && !StringUtils.isEmpty(nonceFromSession)
+                    && nonceFromContext.equals(nonceFromSession)) {
+                validNonceValue = true;
+            }
+            return validNonceValue;
+        } else {
+            // Return true if session nonce cookie config is not enabled.
+            return true;
```
Comment (thanujalk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionNonceCookieUtil.java:99]: Suggestion: if you start the method like
```
if (!isNonceCookieEnabled()) {
   return true;
}

....rest of the logic...
```
It is much cleaner.

**Review by thanujalk — APPROVED** [review]

**Pull request merged**